hail/src/main/scala/is/hail/expr/ir/StringTableReader.scala

```diff
@@ -34,6 +39,80 @@ object StringTableReader {
     status
   }
 }
+case class StringTablePartitionReader(lines: GenericLines) extends PartitionReader{
+  override def contextType: Type = lines.contextType
+
+  override def fullRowType: Type = TStruct("file"-> TString, "text"-> TString)
+
+  override def rowRequiredness(requestedType: Type): TypeWithRequiredness = {
+    val req = BaseTypeWithRequiredness.apply(requestedType).asInstanceOf[RStruct]
+    req.fields.foreach(field => field.typ.hardSetRequiredness(true))
+    req.hardSetRequiredness(true)
+    req
+  }
+
+  override def emitStream(
+     ctx: ExecuteContext,
+     cb: EmitCodeBuilder,
+     context: EmitCode,
+     partitionRegion: Value[Region],
+     requestedType: Type): IEmitCode = {
+
+     context.toI(cb).map(cb) { partitionContext =>
+       val iter = cb.emb.genFieldThisRef[CloseableIterator[GenericLine]]("iter")
+
+       val ctxMemo = partitionContext.asBaseStruct.memoize(cb, "string_table_reader_ctx")
+       val fileName = cb.emb.genFieldThisRef[String]("fileName")
+       val line = cb.emb.genFieldThisRef[String]("line")
```

Comment:
Let's give these generated fields more unique names. Like `stpr_iter` or `string_table_reader_iter` instead of just `iter` (you only have to change the string name, not the scala variable name).